### PR TITLE
feat: update substrate to v1.0.0

### PR DESCRIPTION
**Pull request by ukint-vs**
- https://github.com/paritytech/substrate/pull/14387
- https://github.com/paritytech/substrate/pull/14306
- https://github.com/paritytech/substrate/pull/14437
- https://github.com/paritytech/substrate/pull/14290
- use `into_inner()` in `RuntimeApiImpl::into_parts()`
- `HoldIdentifier` -> `RuntimeHoldReason`
- remove `serde` from `gear-common`

Migrations:
- pallet-im-online v1 https://github.com/paritytech/substrate/pull/14251

@gear-tech/dev

### CODE REVIEW

**Comment by ukint-vs** [comment]
CI - https://github.com/gear-tech/gear/actions/runs/7274623805

**Review by ekovalev — APPROVED** [review]
Great work.

File: pallets/gear-scheduler/src/tests.rs [hunk 1]
```diff
@@ -64,7 +64,7 @@ fn dispatch_from(src: impl Into<ProgramId>) -> StoredDispatch {
 
 fn populate_wl_from(
     src: <Test as frame_system::Config>::AccountId,
-    bn: <Test as frame_system::Config>::BlockNumber,
+    bn: frame_system::pallet_prelude::BlockNumberFor<Test>,
```
Comment (gshep) [inline pallets/gear-scheduler/src/tests.rs:67]: Probably better to import the entity

File: pallets/gear-voucher/src/mock.rs [hunk 2]
```diff
@@ -23,26 +23,22 @@ use frame_support::{
...
     traits::{BlakeTwo256, IdentityLookup},
+    BuildStorage,
 };
 use sp_std::convert::{TryFrom, TryInto};
 
-type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
 type Block = frame_system::mocking::MockBlock<Test>;
 type AccountId = u64;
-type BlockNumber = u64;
+// type BlockNumber = u64;
```
Comment (gshep) [inline pallets/gear-voucher/src/mock.rs:33]: unused?

File: pallets/gear/src/tests.rs [hunk 3]
```diff
@@ -14963,7 +14982,7 @@ mod utils {
     pub(super) type DispatchCustomResult<T> = Result<T, DispatchErrorWithPostInfo>;
     pub(super) type AccountId = <Test as frame_system::Config>::AccountId;
 
-    type BlockNumber = <Test as frame_system::Config>::BlockNumber;
+    // type BlockNumber = u32;
```
Comment (gshep) [inline pallets/gear/src/tests.rs:14985]: unnecessary?

**Review by gshep — APPROVED** [review]
:+1:

File: pallets/gear-scheduler/src/tests.rs (hunk 1 above)
Comment (ukint-vs) [reply pallets/gear-scheduler/src/tests.rs]: done

File: pallets/gear/src/tests.rs (hunk 3 above)
Comment (ukint-vs) [reply pallets/gear/src/tests.rs]: done

**Comment by ukint-vs** [comment]
CI - https://github.com/gear-tech/gear/actions/runs/7288222882

**Comment by ukint-vs** [comment]
CI - https://github.com/gear-tech/gear/actions/runs/7288906444

File: node/authorship/src/tests.rs [hunk 4]
```diff
@@ -231,7 +231,7 @@ macro_rules! init {
         $genesis_hash:ident
     } => {
         let client_builder =
-            TestClientBuilder::new().set_execution_strategy(ExecutionStrategy::NativeWhenPossible);
+            TestClientBuilder::new();
```
Comment (breathx) [inline node/authorship/src/tests.rs:234]:
```suggestion
TestClientBuilder::new();
```

File: node/service/src/chain_spec/vara.rs [hunk 5]
```diff
@@ -702,18 +702,21 @@ fn testnet_genesis(
 
     balances.push((bank_account, EXISTENTIAL_DEPOSIT));
 
-    GenesisConfig {
+    RuntimeGenesisConfig {
         system: SystemConfig {
             // Add Wasm runtime to storage.
             code: wasm_binary.to_vec(),
+            ..Default::default()
```
Comment (breathx) [inline node/service/src/chain_spec/vara.rs:709]: whats the default params?
shouldn't we set them for local node matching real runtime?

File: node/testing/src/genesis.rs [hunk 6]
```diff
@@ -56,19 +56,22 @@ pub fn config_endowed(code: Option<&[u8]>, extra_endowed: Vec<AccountId>) -> Gen
             .map(|endowed| (endowed, 100 * ECONOMIC_UNITS)),
     );
 
-    GenesisConfig {
+    RuntimeGenesisConfig {
         system: SystemConfig {
             code: code
                 .map(|x| x.to_vec())
                 .unwrap_or_else(|| wasm_binary().to_vec()),
+            ..Default::default()
```
Comment (breathx) [inline node/testing/src/genesis.rs:64]: same here

File: pallets/gear/src/lib.rs [hunk 7]
```diff
@@ -472,7 +472,7 @@ pub mod pallet {
     /// May be less than system pallet block number if panic occurred previously.
     #[pallet::storage]
     #[pallet::getter(fn block_number)]
-    pub(crate) type BlockNumber<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+    pub(crate) type BlockNumber<T: Config> = StorageValue<_, BlockNumberFor<T>, ValueQuery>;
```
Comment (breathx) [inline pallets/gear/src/lib.rs:475]: want the underlying type changed?

File: pallets/gear/src/lib.rs [hunk 8]
```diff
@@ -531,10 +531,10 @@ pub mod pallet {
         ///
         /// For tests only.
         #[cfg(any(feature = "std", feature = "runtime-benchmarks", test))]
-        pub fn set_block_number(bn: u32) {
+        pub fn set_block_number(bn: BlockNumberFor<T>) {
             use sp_runtime::SaturatedConversion;
 
-            <BlockNumber<T>>::put(bn.saturated_into::<T::BlockNumber>());
+            <BlockNumber<T>>::put(bn.saturated_into::<BlockNumberFor<T>>());
```
Comment (breathx) [inline pallets/gear/src/lib.rs:537]: no need in the fn tho

File: pallets/gear/src/tests.rs [hunk 9]
```diff
@@ -7362,42 +7382,6 @@ fn pay_program_rent_syscall_works() {
             message_id,
             ActorExecutionErrorReplyReason::Trap(TrapExplanation::Panic(error_text.into())),
         );
-
-        // pay maximum possible rent
```
Comment (breathx) [inline pallets/gear/src/tests.rs:7366]: ?

File: pallets/payment/src/mock.rs [hunk 10]
```diff
@@ -31,18 +31,18 @@ use frame_system as system;
...
 use sp_std::{
     convert::{TryFrom, TryInto},
     prelude::*,
 };
 
-type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
 type Block = frame_system::mocking::MockBlock<Test>;
 type AccountId = u64;
-type BlockNumber = u64;
+type BlockNumber = u32;
```
Comment (breathx) [inline pallets/payment/src/mock.rs:45]: we don't need this type rn?

File: runtime/vara/src/migrations.rs [hunk 11]
```diff
@@ -94,7 +94,9 @@ impl<
 ///
 /// Should be cleared after every release.
 pub type Migrations = (
-    // unreleased
+    // v1030
     UpdatePalletsVersions<Runtime>,
```
Comment (breathx) [inline runtime/vara/src/migrations.rs:98]: these may be removed

**Review by breathx — APPROVED** [review]

File: runtime/vara/src/migrations.rs (hunk 11 above)
Comment (ukint-vs) [reply runtime/vara/src/migrations.rs:98]: CI will broke

**Comment by breathx** [comment]
After node starts
> CLI parameter `--execution` has no effect anymore and will be removed in the future!

File: node/service/src/chain_spec/vara.rs (hunk 5 above)
Comment (ukint-vs) [reply node/service/src/chain_spec/vara.rs:709]: There is just config.
```rust

		#[serde(skip)]
		pub _config: sp_std::marker::PhantomData<T>,
```

Comment (breathx) [reply node/service/src/chain_spec/vara.rs:709]: > There is just config.
> 
> ```rust
> 		#[serde(skip)]
> 		pub _config: sp_std::marker::PhantomData<T>,
> ```

Yeah, just found this out while was looking on substrate changes between branches. I'd manually specify these configs in order not to miss anything in future that would be added into genesis configs and migrations.

**Comment by ukint-vs** [comment]
CI - https://github.com/gear-tech/gear/actions/runs/7337901108

**Comment by ukint-vs** [comment]
CI - https://github.com/gear-tech/gear/actions/runs/7340259444

**Comment by ukint-vs** [comment]
CI - https://github.com/gear-tech/gear/actions/runs/7340306173

**Pull request merged**